noxfile.py

```diff
@@ -0,0 +1,173 @@
+######################################################################
+#
+# File: noxfile.py
+#
+# Copyright 2020 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+import os
+import subprocess
+
+import nox
+
+CI = os.environ.get('CI') is not None
+NOX_PYTHONS = os.environ.get('NOX_PYTHONS')
+
+PYTHON_VERSIONS = ['3.5', '3.6', '3.7', '3.8'] if NOX_PYTHONS is None else NOX_PYTHONS.split(',')
+PYTHON_DEFAULT_VERSION = PYTHON_VERSIONS[-1]
+
+PY_PATHS = ['b2', 'test', 'noxfile.py', 'setup.py']
+
+REQUIREMENTS_FORMAT = ['yapf==0.27']
+REQUIREMENTS_LINT = ['yapf==0.27', 'pyflakes==2.2.0', 'pytest==5.4.3']
+REQUIREMENTS_TEST = ['nose==1.3.7', 'pytest==5.4.3', 'pytest-cov==2.10.0']
+REQUIREMENTS_BUILD = ['liccheck==0.4.7', 'setuptools>=20.2']
+REQUIREMENTS_DOC = [
+    'sphinx', 'sphinx-autobuild', 'sphinx_rtd_theme', 'sphinx-argparse', 'sphinxcontrib-plantuml',
+    'sadisplay'
+]
+
+nox.options.reuse_existing_virtualenvs = True
+nox.options.sessions = [
+    'lint',
+    'test',
+]
+
+# In CI, use Python interpreter provided by GitHub Actions
+if CI:
+    nox.options.force_venv_backend = 'none'
+
+
+def install_myself(session):
+    """Install from the source."""
+    # In CI, install B2 SDK from the master branch
+    if CI:
+        session.install('git+git://github.com/Backblaze/b2-sdk-python#egg=b2sdk')
+
+    session.install('-e', '.')
+
+
+# noinspection PyShadowingBuiltins
+@nox.session(python=PYTHON_DEFAULT_VERSION)
+def format(session):
```

Comment:
redefining builtin "format"

Reply:
Yeap, I know :D and I can live with that :D
However, I can change to sth like this if you like it better:
```python
@nox.session(name='format', python=PYTHON_DEFAULT_VERSION)
def format_(session):
   ...
```

Reply:
yes please, if only for Codacy to leave us alone

Reply:
Done